main/SQLHelper.cpp

```diff
@@ -13,6 +13,8 @@
 #include "../sqlite/sqlite3.h"
 #endif
 #include "../hardware/hardwaretypes.h"
+#include "../hardware/blebox/db/migration.h"
```

Comment:
As the 'migration' is only done once and there is not much code, please do it in this file.
No need for a separate migration ...

Reply:
- I moved the migration to an existing file (BleBox.cpp) in 2216906be

Of course, putting the whole migration inline wouldn't make sense, because SQLHelper.cpp would have to require a lot more headers.

Ideally, migrations should be separate from "SQL helpers" anyway, along with the schema. And since migrations can get messy in production code, one-file-per-migration is a good approach, especially since older migrations should be periodically deprecated/removed anyway. (And simply removing files from a repository is the best way to remove functionality).

I'm getting this sens of "allergy" towards having "lots of files", which makes no sense in 2020, especially with git, modern filesystems and the need to reduce maintenance burdens. (Instead of having gigantic files with enormous switch/if statements).

So I really don't understand what you mean here and what benefit outweighs the costs.